spec/controllers/sessions_controller_spec.rb

```diff
@@ -0,0 +1,75 @@
+require 'spec_helper'
+
+describe SessionsController do
+  describe "GET new" do
+    it "redirects to home page for authenticated users" do
+      session[:user_id] = Fabricate(:user).id
+      get :new
+      expect(response).to redirect_to(home_path)
+    end
+
+    it "render new template for unauthenticated users" do
+      get :new
+      expect(response).to render_template(:new)
+    end
+  end
+
+  describe "POST create" do
+    let(:ellery) { Fabricate(:user) }
+
+    context "with valid credentials" do
+      before do
+        post :create, email: ellery.email, password: ellery.password
+      end
+
+      it "puts the signed in user in the session" do
+        expect(session[:user_id]).to eq(ellery.id)
+      end
+
+      it "redirects user to the home page" do
+        expect(response).to redirect_to(home_path)
+      end
+
+      it "sets the notice" do
+        expect(flash[:notice]).not_to be_blank
+      end
+    end
+
+    context "with invalid credentials" do
+      before do
+        post :create, email: ellery.email, password: ellery.password + 'asf'
```

Comment:
For wrong password I would make it less complicated. `"wrong"` or something obvious - no need to concatenate with real pw.